Updates from ESRR feedback (mostly FAQs) (#235)
nasa-fornax/fornax-documentation: documentation/faq.md

# Frequently Asked Questions (FAQ)

## Can I run high-performance computing (HPC) workflows?

Although the Fornax Science Console isn't a traditional HPC system, it provides many similar features and benefits.

> By selecting a very large [server size](#server-and-env-options), users can run highly parallelized workflows on a single node and achieve performance that often eliminates the need for a multi-node HPC system.

Reviewer: ```suggestion
By selecting a very large [server size](#server-and-env-options), users can run highly parallelized workflows that replicate functionality of a multi-node HPC system.
```

Author: Fornax can't exactly replicate the functionality of a multi-node HPC which is why I was more specific with the wording. If you let me know what's confusing or awkward about the current sentence I can try to fix it.

> In addition to the substantial compute power for data *processing*, the high network bandwidth enables massively parallel *access* to [cloud-hosted data](#cloud-hosted-data).

Reviewer: turn the sentence around to make the parallel data access the point.  
```suggestion
Massively parallel data *access* to [cloud-hosted data](#cloud-hosted-data) is enabled by high network bandwidth between Fornax and some NASA datasets.
```

I would also suggest a bullet point about fast data access to cloud datasets since both Fornax and those datasets are in the same AWS cloud.

We do plan to add support for batch or asynchronous job submission in a future release.

Be aware that server sessions can be automatically terminated under certain circumstances.
See [](#jupyterlab-session-information) for details.

## What are the pre-installed directories and files used for?

You can store files and data in the directories `~/` (home directory), `~/s3-storage`, `~/shared-storage`, and `/scratch`.
See [](#data-storage) for details.

The directory `~/fornax-notebooks` contains demo and tutorial notebooks.
It is read-only.
See [](#notebooks-in-fornax) for details.

`$ENV_DIR` and `$USER_ENV_DIR` are aliases to directories where {term}`environments <Environment>` are stored.
`$LOCK_DIR` is an alias to a directory where you'll find lists of the software installed in the built-in environments.
See [](#compute-environments) for details.

The file `~/.profile` is an initialization script that runs when a {term}`terminal <Terminal>` session starts.
See [](#terminal-initialization) for details.

## What software is pre-installed?

[Software Overview](#software-overview) lists notable packages that are pre-installed.

> These and many other pre-installed packages are organized into several environments.

Reviewer: ```suggestion
A software environment is a self-contained workspace that bundles a specific version of a programming language and all the libraries or tools a piece of code depends on, ensuring that code is reproducible.  
Fornax has several [software environments](#software-overview) available for users.
```

The word 'packages' has too many meanings.  I think it is better to define what an environment is in a FAQ before using it.  Then link to the two pages.

Author: We have a definition of "environment" in the glossary so I'll link to that. I think on this site we're only using "packages" to mean software packages. What other meanings are you seeing or thinking of?

[](#preinstalled-software) describes how to find complete lists of the packages in each environment and how to search for a particular package of interest.

## How do I install compilers or other packages that require root access?

> We recommend installing these packages from conda, which avoids the need for root access.

Reviewer: ```suggestion
We recommend installing these packages from conda, using miniconda, which avoids the need for root access.
```

Fornax is also co-located with NASA archive data on the cloud, making it easy to pull large datasets without the bottleneck of downloading to a local machine.
Cloud-hosted NASA mission data are readily accessible at high bandwidth directly from the Fornax Science Console.

(how-to-upload-download)=
## How do I upload or download files and directories?

:::{tip}
By accessing NASA Astrophysics data through [APIs](#apis) (small to medium amounts of data) or reading it directly from [NASA's ODR buckets](#cloud-hosted-data) (unlimited), you can avoid having to upload it to the Fornax Science Console first.

> We also recommend that you minimize the amount of data you download because your [credits](#intro-best-practices) are charged for data egress.

Reviewer: ```suggestion
Note that [credits](#intro-best-practices) will be charged for data download out of Fornax (known as "data egress") 
```

:::

> There are two ways to move files in and out of the Fornax Science Console: upload and download to your local computer, or use an API to connect with an external storage solution.

Reviewer: you discuss these below in the opposite order.  Also I think the use of the term "API" here is unnecessary to making this clear.  If you keep it, you need to define it without assuming people read the tip above.  

```suggestion
There are two ways to move files in and out of the Fornax Science Console:  connect with an external storage solution, or upload and download to your local computer
```
